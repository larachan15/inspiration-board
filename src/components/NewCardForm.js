import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';
import './NewCardForm.css';

const EMOJI_LIST = ["", "heart_eyes", "beer", "clap", "sparkling_heart", "heart_eyes_cat", "dog", "relaxed", "balloon", "black_heart"]


class NewCardForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      text: "",
      emoji: ""
    };
  }

  onFormChange = (event) => {
    console.log("Made a change!")
    const field = event.target.name;
    const value = event.target.value;

    const updatedState = {};
    updatedState[field] = value;
    this.setState(updatedState)
  }

  onFormSubmit = (event) => {
    event.preventDefault();

    const newCard = {
      card: {
        text: this.state.text,
        emoji: this.state.emoji
      }
    }

    this.setState({
      text: '',
      emoji: ''
    });

    console.log("successfully created a new card whoo!", newCard);
    this.props.addCardCallback(newCard);
  }

  emojiSelection = () => {
   return  EMOJI_LIST.map((item, i) => {
     return <option
       key={i}
       value={item}>{emoji.getUnicode(item)}</option>
   });
 }

  render() {
    return (
      <div className="new-card-form">
        <div className="new-card-form__header">
          <h2>New Card</h2>
        </div>

        <form className="new-card-form__form"
              onSubmit={this.onFormSubmit}>
              <label htmlFor="text">Your Message</label>
              <textarea
                className="new-card-form__form-textarea"
                name="text"
                value={this.state.text}
                onChange={this.onFormChange}
              />
            <label htmlFor="emoji">Emoji</label>
            <select
              className="new-card-form__form-select"
              name="emoji"
              value={this.state.emoji}
              onChange={this.onFormChange}>
              {this.emojiSelection()}
            </select>
            <input type="submit"
                   value="Submit Message"
                   className="new-card-form__form-button"
            />
        </form>
      </div>
    )
  }

}

export default NewCardForm;
