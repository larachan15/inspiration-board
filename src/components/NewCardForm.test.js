import React from 'react';
import NewCardForm from './NewCardForm';
import { shallow } from 'enzyme';

describe('NewCardForm', () => {
  test('that it matches an existing snapshot for adding', () => {
    const wrapper = shallow(
      <NewCardForm  />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
