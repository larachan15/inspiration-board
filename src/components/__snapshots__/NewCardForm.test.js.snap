// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewCardForm that it matches an existing snapshot for adding 1`] = `
<div
  className="new-card-form"
>
  <div
    className="new-card-form__header"
  >
    <h2>
      New Card
    </h2>
  </div>
  <form
    className="new-card-form__form"
    onSubmit={[Function]}
  >
    <label
      htmlFor="text"
    >
      Your Message
    </label>
    <textarea
      className="new-card-form__form-textarea"
      name="text"
      onChange={[Function]}
      value=""
    />
    <label
      htmlFor="emoji"
    >
      Emoji
    </label>
    <select
      className="new-card-form__form-select"
      name="emoji"
      onChange={[Function]}
      value=""
    >
      <option
        key="0"
        value=""
      />
      <option
        key="1"
        value="heart_eyes"
      >
        😍
      </option>
      <option
        key="2"
        value="beer"
      >
        🍺
      </option>
      <option
        key="3"
        value="clap"
      >
        👏
      </option>
      <option
        key="4"
        value="sparkling_heart"
      >
        💖
      </option>
      <option
        key="5"
        value="heart_eyes_cat"
      >
        😻
      </option>
      <option
        key="6"
        value="dog"
      >
        🐶
      </option>
      <option
        key="7"
        value="relaxed"
      >
        ☺️
      </option>
      <option
        key="8"
        value="balloon"
      >
        🎈
      </option>
      <option
        key="9"
        value="black_heart"
      >
        🖤
      </option>
    </select>
    <input
      className="new-card-form__form-button"
      type="submit"
      value="Submit Message"
    />
  </form>
</div>
`;
